docs/bugs_and_glitches.md

```diff
@@ -290,21 +273,36 @@ This is a bug with `CheckPlayerHasUsableMoves` in [engine/battle/core.asm](/engi
 
 This bug affects Attract, Curse, Foresight, Mean Look, Mimic, Nightmare, Spider Web, Transform, and stat-lowering effects of moves like String Shot or Bubble during the semi-invulnerable turn of Fly or Dig.
 
-This is a bug with `CheckHiddenOpponent` in [engine/battle/effect_commands.asm](/engine/battle/effect_commands.asm):
+**Fix:** Edit `CheckHiddenOpponent` in [engine/battle/effect_commands.asm](/engine/battle/effect_commands.asm):
+
+```diff
+ CheckHiddenOpponent:
+-; BUG: This routine is completely redundant and introduces a bug, since BattleCommand_CheckHit does these checks properly.
+-	ld a, BATTLE_VARS_SUBSTATUS3_OPP
+-	call GetBattleVar
+-	and 1 << SUBSTATUS_FLYING | 1 << SUBSTATUS_UNDERGROUND
+ 	ret
+```
 
+**Alternate fix:**
```

Comment:
I honestly think this entire section is redundant. There's always better and more efficient ways of fixing a bug, no need to list them all. Either remove this solution or remove the sloppy solution.
Now I think about it, wouldn't it be a good idea to `and a` before returning in `CheckHiddenOpponent`? Just to make absolutely sure the zero flag is unset. Not all moves and possible entrypoints for `CheckHiddenOpponent` have been tested.

Reply:
Agreed about both ways being unnecessary.

If ~~xor a~~ `and a` isn't necessary in `CheckHiddenOpponent`, IMO the fix shouldn't have it. If it *is* necessary, then the current fix isn't a fix at all, which is what testing was supposed to catch. (And yeah, if testing hasn't actually verified that the current fix works, then the `and a` is probaby necessary.)